lib/fizzbuzz.rb

```diff
@@ -1,9 +1,17 @@
 class SuperFizzBuzz
 
   def run(input)
-
-    #Implement your code here
-
+  
+	  case 
+	  when input % 3 == 0 && input % 5 != 0
+	  	 "Fizz"
+	  when input % 5 == 0 && input % 3 != 0
+	  	 "Buzz"
+	  when input % 3 == 0 && input % 5 == 0
+	  	 "FizzBuzz"
+	  else
+	  	 input
+	  end
   end
```

Comment:
I think this could be substantially improved by ordering your `when` conditions better. The way that a case structure works, it will return whichever when block is satisfied first and then stop checking. So instead of checking if it's divisible by 3 but not 5, then if it's divisible by 5 but not 3, and then finally if it's divisible by both, you could do your check for divisibility by both first and that would greatly simplify the other two checks.

e.g.:

`when input % 3 == 0 and input % 5 == 0 ...
when input % 5 == 0 ...
when input % 3 == 0 ...
else ...`

I also think that since the blocks being executed under each `when` are so small, it would make sense to use the one-liner `when [conditional] then [code]` syntax to make the overall function smaller & improve readability.